sql/core/src/main/scala/org/apache/spark/sql/GroupedDataset.scala

```diff
@@ -88,6 +104,78 @@ class GroupedDataset[K, T] private[sql](
       MapGroups(f, groupingAttributes, logicalPlan))
   }
 
+  // To ensure valid overloading.
+  def agg(expr: Column, exprs: Column*): DataFrame = groupedData.agg(expr, exprs: _*)
```

Comment:
I'll make it protected.  I want to keep it incase we decide to move forward with the combining, so that we are sure we won't have invalid overloads.